lib/media.js

```diff
@@ -57,50 +52,106 @@ function findTop(str) {
  * @returns {string}
  */
 function splice(str, index, items) {
-  return str.substr(0, index) + _.flattenDeep(items).join('\n') + str.substr(index);
+  return str.substr(0, index) + items + str.substr(index);
+}
+
+/**
+ * Get the contents of a string that come after specified characters.
+ *
+ * @param  {string} str Any string to split
+ * @param  {string} dir The directory the file is in
+ * @return {string}
+ */
+function getFileName(str, dir) {
+  return str.split(dir)[1];
+}
+
+/**
+ * Retrieve the contents of a file
+ *
+ * @param  {array} fileArray   An array of file names
+ * @param  {string} targetDir  The directory to retrieve files from
+ * @param  {string} filePrefix The string that comes right before the file name
+ * @return {Promise}
+ */
+function getContentsOfFiles(fileArray, targetDir, filePrefix) {
+  var allPromises = [],
+    rootDir = process.cwd();
+
+  fileArray.forEach(file => {
+    allPromises.push(files.readFilePromise(path.join(rootDir, targetDir, getFileName(file, filePrefix))));
+  });
+
+  return Promise.all(allPromises)
+    .catch(err => {
+      throw err;
+    });
+}
+
+/**
+ * Concatenate all the styles together in one `<style>` tag
+ * @param  {array} stylesArr The array of all the css files needed for a page
+ * @return {Promise | boolean}
+ */
+function combineStyles(stylesArr) {
+  if (!stylesArr || !stylesArr.length > 0) {
+    return false;
+  }
+
+  // If there are styles, retrieve contents
+  return getContentsOfFiles(stylesArr, 'public/css', '/css/')
+    .then(data => '<style>' + data.join('') + '</style>');
+}
+
+/**
+ * Concatenate all the scripts together in one `<script>` tag
+ *
+ * @param  {array} scriptsArr The array of all the script files needed for a page
+ * @return {Promise | boolean}
+ */
+function combineScripts(scriptsArr) {
+  if (!scriptsArr || !scriptsArr.length) {
```

Comment:
- Need to remove the `> 0`. Doesn't break, but unnecessary

- I considered merging them into one, but I also kind of like the idea of keeping style and script concatenation along two separate paths incase we ever want to perform secondary operations on one of them. I was thinking about how we're going to end up injecting scripts for one-off interactives and it might be necessary to further modify the scripts...? Maybe I'm thinking too much into it and that'll be another task. Thoughts?